mvp-spec/partials-updater/src/interpretor.ts

```diff
@@ -57,87 +61,67 @@ export async function interpret(tokens: LexerToken[]): Promise<string> {
-    const jsonStr = token.text
-                    .slice(PartialBeginStartKey.length)
-                    .slice(0, -PartialBeginEndKey.length);
+async function interpretPartialBeginToken(token: LexerToken, lineBreak: string): Promise<string> {
+    const jsonStr = token.text.match(PartialBegin)[1];
     try {
         const config: PartialConfig = JSON.parse(jsonStr);
-        if (!config.hasOwnProperty('files') && config.files.length >= 1) {
-            throw new Error(`Partial with bad configuration: ${token.text}`)
+        if (config.files?.length < 1) {
+            throw ''
         }
-        
+
         if (config.jq !== undefined) {
             const jsonPaths = config.files.map(getPartialPath);
-            const options = (config.files.length > 1) ? { slurp: true } : undefined;
+            const options = (config.files.length > 1) ? {slurp: true} : undefined;
             const json = (await jq.run(config.jq, jsonPaths, options)).toString();
-            const partial = getPartialInCodeBlock(json);
-            return partial;
-        } 
-        
+            return getPartialInCodeBlock(json, lineBreak);
+        }
+
         if (config.transformation !== undefined) {
             if (config.transformation == "mermaid") {
-                return transformMermaids(config);
+                return transformMermaids(config, lineBreak);
             } else {
                 throw new Error(`Unknown type of Transformation: ${config.transformation}`);
             }
         }
-        
-        const partials = await Promise.all(config.files.map(loadPartial));
-        const lineBreak = getLineBreakSymbol(config.files[0]);
-        let merged = partials.join(lineBreak)
-        if (config.block !== undefined) {
-            merged = getPartialInCodeBlock(merged, config.block);
-        }
-        return merged;
+
+        const partials: Document[] = await Promise.all(config.files.map(loadPartial));
+
+        const merged = partials.map(d => d.content).join(lineBreak)
+
+        return config.block ? getPartialInCodeBlock(merged, lineBreak, config.block) : merged;
     } catch (e) {
         throw new Error(`Partial with bad configuration: ${token.text} - ${e}`)
     }
 }
 
-async function transformMermaids(config: PartialConfig) {
-    const lineBreak = getLineBreakSymbol(config.files[0]);
-    const cmdAndPathes = config.files.map(buildMermaidCommand);
-    cmdAndPathes.forEach((val) => {
+async function transformMermaids(config: PartialConfig, lineBreak: string) {
+    const cmdAndPaths = config.files.map(buildMermaidCommand);
+    for (const val of cmdAndPaths) {
         child.execSync(val.cmd);
-    });
-    const contents = cmdAndPathes.map((val) => {
+        // Remove id from generated file
+        const generatedPath = getAssetPath(val.assetFile);
+        const generatedDoc = await loadFile(generatedPath)
+        await rewriteFile(generatedPath, generatedDoc.content.replace(/^<svg id\=\"[^"]*"/, "<svg "))
```

Comment:
I guess this is another subject of the EOL fix and maybe we should address it in another PR.

Moreover, if I understood well, it doesn't fix the issue that we have with the comparations of generated documents.

Reply:
Yes. Will revert this part.